synapse/storage/database.py

```diff
@@ -1268,6 +1268,7 @@ async def simple_upsert_many(
         value_names: Collection[str],
         value_values: Collection[Collection[Any]],
         desc: str,
+        lock: bool = True,
```

Comment:
Note that I *think* this is fine to default to `True` since `simple_upsert_txn_emulated` (which this eventually calls) defaults to `True`. So pretty much this was the default behavior.